compiler/src/iree/compiler/GlobalOptimization/Passes.cpp

```diff
@@ -51,11 +51,17 @@ static llvm::cl::opt<bool> clEnableFuseHorizontalContractions(
         "Enables horizontal fusion of contractions with one common operand"),
     llvm::cl::init(false));
 
-static llvm::cl::opt<bool> clEnableDemoteContractionInputsToBF16(
+static llvm::cl::opt<DemotionOption> clDemoteContractionInputsToBF16Strategy(
     "iree-global-opt-enable-demote-contraction-inputs-to-bf16",
-    llvm::cl::desc(
-        "Demote inputs (LHS, RHS) of linalg matmul-like ops from f32 to bf16."),
-    llvm::cl::init(false));
+    llvm::cl::desc("Demotes inputs (LHS, RHS) of contraction ops to BF16. "
+                   "Selects types of contraction ops to demote."),
+    llvm::cl::values(
+        clEnumValN(DemotionOption::All, "all", "Demote all contraction ops."),
+        clEnumValN(DemotionOption::Conv, "conv",
+                   "Only demote convolution ops."),
+        clEnumValN(DemotionOption::Matmul, "matmul", "Only demote matmul ops."),
+        clEnumValN(DemotionOption::None, "none", "Demote no contraction ops.")),
+    llvm::cl::init(DemotionOption::All));
```

Comment:
hmm, I think the default value should be `None`? It is an experimental flag, so we don't want to it be on by default.

Reply:
Yes. Done.